<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1509</width>
    <height>609</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Algorithm Visualizer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>1491</width>
      <height>561</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,2,0">
     <property name="spacing">
      <number>3</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SizeConstraint::SetMinimumSize</enum>
     </property>
     <property name="leftMargin">
      <number>3</number>
     </property>
     <property name="topMargin">
      <number>3</number>
     </property>
     <property name="rightMargin">
      <number>3</number>
     </property>
     <property name="bottomMargin">
      <number>3</number>
     </property>
     <item>
      <widget class="QScrollArea" name="LHS">
       <property name="minimumSize">
        <size>
         <width>160</width>
         <height>320</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>400</width>
         <height>1280</height>
        </size>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>398</width>
          <height>553</height>
         </rect>
        </property>
        <widget class="QPushButton" name="load_map">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>350</y>
           <width>80</width>
           <height>24</height>
          </rect>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>load map</string>
         </property>
        </widget>
        <widget class="QPushButton" name="clear">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>210</y>
           <width>81</width>
           <height>31</height>
          </rect>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Reset</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>170</y>
           <width>81</width>
           <height>31</height>
          </rect>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Start/Pause</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>110</y>
           <width>131</width>
           <height>20</height>
          </rect>
         </property>
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Visualiser</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>170</y>
           <width>131</width>
           <height>31</height>
          </rect>
         </property>
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Start/Pause Algorithm</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>210</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Reset Algorithm</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="Gsize_line_edit">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>330</y>
           <width>121</width>
           <height>24</height>
          </rect>
         </property>
         <property name="readOnly">
          <bool>false</bool>
         </property>
        </widget>
        <widget class="QLabel" name="prop1">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>330</y>
           <width>61</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Grid Size</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="Ssize_line_edit">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>360</y>
           <width>121</width>
           <height>24</height>
          </rect>
         </property>
         <property name="readOnly">
          <bool>false</bool>
         </property>
        </widget>
        <widget class="QLabel" name="prop1_2">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>360</y>
           <width>81</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Square Size</string>
         </property>
        </widget>
        <widget class="QPushButton" name="Update_Map_button">
         <property name="geometry">
          <rect>
           <x>80</x>
           <y>400</y>
           <width>141</width>
           <height>21</height>
          </rect>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>Update Map</string>
         </property>
        </widget>
        <widget class="Line" name="line_3">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>271</y>
           <width>431</width>
           <height>21</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>290</y>
           <width>71</width>
           <height>20</height>
          </rect>
         </property>
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Map Settings</string>
         </property>
        </widget>
        <widget class="QLabel" name="prop1_3">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>431</y>
           <width>61</width>
           <height>21</height>
          </rect>
         </property>
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Steps/sec</string>
         </property>
        </widget>
        <widget class="QSpinBox" name="spinBox">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>434</y>
           <width>61</width>
           <height>21</height>
          </rect>
         </property>
         <property name="value">
          <number>1</number>
         </property>
        </widget>
        <widget class="QFrame" name="frame_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>281</width>
           <height>71</height>
          </rect>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Raised</enum>
         </property>
         <widget class="QComboBox" name="Algorithm_dropdown">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>30</y>
            <width>181</width>
            <height>24</height>
           </rect>
          </property>
          <item>
           <property name="text">
            <string>Choose Algorithm </string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>A*</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Dijkstra</string>
           </property>
          </item>
         </widget>
         <widget class="QLabel" name="label">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>10</y>
            <width>121</width>
            <height>20</height>
           </rect>
          </property>
          <property name="autoFillBackground">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Choose Algorithm </string>
          </property>
         </widget>
        </widget>
        <widget class="Line" name="line_4">
         <property name="geometry">
          <rect>
           <x>110</x>
           <y>300</y>
           <width>91</width>
           <height>31</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
        <widget class="Line" name="line_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>120</y>
           <width>81</width>
           <height>31</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
        <widget class="Line" name="line_6">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>80</y>
           <width>401</width>
           <height>31</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>470</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>set start square</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_3">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>470</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>set end square</string>
         </property>
        </widget>
        <widget class="Line" name="line_7">
         <property name="geometry">
          <rect>
           <x>-20</x>
           <y>450</y>
           <width>421</width>
           <height>21</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_4">
         <property name="geometry">
          <rect>
           <x>280</x>
           <y>470</y>
           <width>101</width>
           <height>31</height>
          </rect>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>set obstacle</string>
         </property>
        </widget>
        <widget class="Line" name="line_8">
         <property name="geometry">
          <rect>
           <x>-20</x>
           <y>500</y>
           <width>421</width>
           <height>21</height>
          </rect>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGraphicsView" name="graphicsView">
       <property name="minimumSize">
        <size>
         <width>320</width>
         <height>320</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>800</width>
         <height>800</height>
        </size>
       </property>
       <property name="cursor" stdset="0">
        <cursorShape>CrossCursor</cursorShape>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frame">
       <property name="minimumSize">
        <size>
         <width>160</width>
         <height>320</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>400</width>
         <height>1280</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::Shape::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Shadow::Raised</enum>
       </property>
       <widget class="QLabel" name="label_6">
        <property name="geometry">
         <rect>
          <x>90</x>
          <y>20</y>
          <width>101</width>
          <height>51</height>
         </rect>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>Algorithm Details</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEdit">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>110</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>100</y>
          <width>71</width>
          <height>21</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::Shape::NoFrame</enum>
        </property>
        <property name="text">
         <string>Name</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>140</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::Shape::NoFrame</enum>
        </property>
        <property name="text">
         <string>Steps Taken</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEdit_2">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>140</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLabel" name="label_9">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>170</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::Shape::NoFrame</enum>
        </property>
        <property name="text">
         <string>Nodes Visited</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEdit_3">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>170</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QLabel" name="label_10">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>200</y>
          <width>81</width>
          <height>21</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::Shape::NoFrame</enum>
        </property>
        <property name="text">
         <string>Execution Time</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEdit_4">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>200</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1509</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menufile">
    <property name="title">
     <string>file</string>
    </property>
    <addaction name="actionload"/>
    <addaction name="actionsave"/>
    <addaction name="separator"/>
    <addaction name="actionexit"/>
   </widget>
   <widget class="QMenu" name="menuoptions">
    <property name="title">
     <string>options</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuhelp">
    <property name="title">
     <string>help</string>
    </property>
   </widget>
   <addaction name="menufile"/>
   <addaction name="menuoptions"/>
   <addaction name="menuhelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionload">
   <property name="text">
    <string>load...</string>
   </property>
  </action>
  <action name="actionsave">
   <property name="text">
    <string>save</string>
   </property>
  </action>
  <action name="actionexit">
   <property name="text">
    <string>exit</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
